mtags/src/main/scala-3/scala/meta/internal/pc/PcCollector.scala

```diff
@@ -383,6 +383,21 @@ abstract class PcCollector[T](driver: InteractiveDriver, params: OffsetParams):
     trees.collect { case t: Tree =>
       t
     }
+
+  // NOTE: Connected to https://github.com/lampepfl/dotty/issues/16771
+  // `sel.nameSpan` is calculated incorrectly in (1 + 2).toString
+  // See test DocumentHighlightSuite.select-parentheses
+  private def selectNameSpan(sel: Select)(using Context): Span =
+    val span = sel.span
+    val realName = sel.name.stripModuleClassSuffix.lastPart
+    if span.exists then
+      val point = span.point
+      if sel.name.toTermName == nme.ERROR then Span(point)
+      else if sel.qualifier.span.start - span.start >= realName.length
```

Comment:
This method is a copy of `sel.nameSpan`, but here we check if 
`sel.qualifier.span.start - span.start >= realName.length`, so `name` would fit as infix operator. This is still not perfect for operations nested in multiple parentheses, but should work for simple cases